connection.go

```diff
@@ -144,6 +144,10 @@ func (mc *mysqlConn) interpolateParams(query string, args []driver.Value) (strin
 	buf = buf[:0]
 	argPos := 0
 
+	if len(args) != strings.Count(query, "?") {
+		return "", ErrArgsInvalid
```

Comment:
I meant `?` may be not a placeholder when it appeared after backslash.
We don't have complete SQL parser MySQL has.